src/vc_util/src/lib.rs

```diff
@@ -1007,4 +1056,165 @@ mod tests {
         );
         assert_matches!(result, Err(e) if format!("{:?}", e).contains("inconsistent claim in VC"));
     }
+
+    fn credential_spec_with_0_args() -> CredentialSpec {
+        CredentialSpec {
+            credential_type: "vcWithoutArgs".to_string(),
+            arguments: None,
+        }
+    }
+
+    fn credential_spec_with_1_arg() -> CredentialSpec {
+        let mut args = HashMap::new();
+        args.insert(
+            "firstArg".to_string(),
+            ArgumentValue::String("string arg value".to_string()),
+        );
+        CredentialSpec {
+            credential_type: "vcWithOneArg".to_string(),
+            arguments: Some(args),
+        }
+    }
+
+    fn credential_spec_with_2_args() -> CredentialSpec {
+        let mut args = HashMap::new();
+        args.insert(
+            "firstArg".to_string(),
+            ArgumentValue::String("string arg value".to_string()),
+        );
+        args.insert("secondArg".to_string(), ArgumentValue::Int(42));
+        CredentialSpec {
+            credential_type: "vcWithOneArg".to_string(),
+            arguments: Some(args),
+        }
+    }
+
+    fn credential_specs_for_test() -> Vec<CredentialSpec> {
```

Comment:
If you were to go one step further I would suggest using `proptest` 😉

Reply:
A good idea... for another step  😉 
(I believe we don't use `proptest` in II yet, do we?)